File: LeetCode/cpp/129.cpp
class Solution {
public:
    struct NodeValue {
        TreeNode *node;
        int value;

        NodeValue(TreeNode *node, int value) : node(node), value(value) {}
    };

    int sumNumbers(TreeNode *root) {
        if (!root)
            return 0;
        int sum = 0;
        queue<NodeValue> que;
        que.push(NodeValue(root, 0));
        while (!que.empty()) {
            NodeValue node = que.front();
            que.pop();
            if (!node.node->left && !node.node->right) {
                sum += node.value * 10 + node.node->val;
            }
            if (node.node->left)
                que.push(NodeValue(node.node->left, node.value * 10 + node.node->val));
            if (node.node->right)
                que.push(NodeValue(node.node->right, node.value * 10 + node.node->val));
        }
        return sum;
    }
};

File: LeetCode/python3/129.py
class Solution:
    total = 0
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        self.CountSum(root, 0)
        return self.total


    def CountSum(self, root, sum):
        if not root.left and not root.right:
            self.total += sum * 10 + root.val
            return
        if root.left:
            self.CountSum(root.left, sum * 10 + root.val)
        if root.right:
            self.CountSum(root.right, sum * 10 + root.val)